Source/Core/Core/AchievementManager.cpp

```diff
@@ -855,4 +1124,31 @@ AchievementManager::ResponseType AchievementManager::Request(
   }
 }
 
+AchievementManager::ResponseType
+AchievementManager::RequestImage(rc_api_fetch_image_request_t rc_request, Badge* rc_response)
+{
+  rc_api_request_t api_request;
+  Common::HttpRequest http_request;
+  if (rc_api_init_fetch_image_request(&api_request, &rc_request) != RC_OK)
+  {
+    ERROR_LOG_FMT(ACHIEVEMENTS, "Invalid request for image.");
+    return ResponseType::INVALID_REQUEST;
+  }
+  auto http_response = http_request.Get(api_request.url);
+  if (http_response.has_value() && http_response->size() > 0)
+  {
+    rc_api_destroy_request(&api_request);
+    *rc_response = http_response.value();
+    return ResponseType::SUCCESS;
+  }
+  else
+  {
+    WARN_LOG_FMT(ACHIEVEMENTS,
+                 "RetroAchievements connection failed on image request.\n URL: {}\n Post data: {}",
+                 api_request.url, api_request.post_data);
```

Comment:
`post_data` can be null here, which triggers an assert in fmt. Please check for that.

Reply:
Ahh whoops, this is a GET not a POST so post_data will _always_ be null.